inbox/api/jsc_api.py

```diff
@@ -118,3 +123,63 @@ def auth_callback():
         })
 
         return make_response( (resp, 201, { 'Content-Type': 'application/json' }))
+
+@app.route('/create_account', methods=['POST'])
+def create_account():
+    g.parser.add_argument('target', type=int, location='args')
+    g.parser.add_argument('email', required=True, type=bounded_str, location='form')
+    g.parser.add_argument('smtp_host', required=True, type=bounded_str, location='form')
+    g.parser.add_argument('smtp_port', type=int, location='form')
+    g.parser.add_argument('smtp_username', required=True, type=bounded_str, location='form')
+    g.parser.add_argument('smtp_password', required=True, type=bounded_str, location='form')
+    g.parser.add_argument('imap_host', required=True, type=bounded_str, location='form')
+    g.parser.add_argument('imap_port', type=int, location='form')
+    g.parser.add_argument('imap_username', required=True, type=bounded_str, location='form')
+    g.parser.add_argument('imap_password', required=True, type=bounded_str, location='form')
+    g.parser.add_argument('ssl_required', required=True, type=bool, location='form')
+
+    args = strict_parse_args(g.parser, request.args)
+    shard = (args.get('target') or 0) >> 48
+
+    with session_scope(shard) as db_session:
+        account = db_session.query(Account).filter_by(email_address=args['email']).first()
+
+        provider_auth_info = dict(provider='custom',
+                                  email=args['email'],
+                                  imap_server_host=args['imap_host'],
+                                  imap_server_port=(args.get('imap_port') or DEFAULT_IMAP_SSL_PORT),
```

Comment:
Did it that way because `get()` only uses default value if key doesn't exist. Ex: 
```python
x = {'key': None}
x.get('key', 'default') # => None
```
And that was happening when I was testing the endpoint. I think `g.parser` adds the key to the dictionary even if the param was not present on the request.